### PR TITLE
X10+TensorFlow

### PR DESCRIPTION
**Pull request by compnerd**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Review by pschuh — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

File: utils/swift_build_support/swift_build_support/products/tensorflow.py [hunk 1]
```diff
@@ -112,7 +132,12 @@ def build(self, host_target):
                 '-D', 'BUILD_X10={}'.format(
                     'YES' if self.args.enable_x10 else 'NO'
                 ),
-                '-D', 'X10_INCLUDE_DIR={}'.format(x10_inc),
+                '-D', 'X10_INCLUDE_DIR={}'.format(
+                    _x10_include_directory(host_target, tensorflow_source_dir)
+                ),
+                '-D', 'X10_LIBRARY={}'.format(
```
Comment (asuhan) [inline utils/swift_build_support/swift_build_support/products/tensorflow.py]: Where is `X10_LIBRARY` used? I haven't seen any PR to swift-apis to reflect this.

**Review by asuhan — CHANGES_REQUESTED** [review]
I don't understand how we're using `X10_LIBRARY`. Also, trying to run `ops_test` complains about an undefined symbol (`_ZN9swift_xla10XlaHelpers19s_mat_mult_precisionE`).

File: utils/swift_build_support/swift_build_support/products/tensorflow.py (hunk 1 above)
Comment (compnerd) [reply utils/swift_build_support/swift_build_support/products/tensorflow.py]: This is a staging change, to pass the variable, that will be used subsequently.  Its just passing an extra variable and adds the additional binary to the image.

Comment (asuhan) [reply utils/swift_build_support/swift_build_support/products/tensorflow.py]: Ok, my point about `ops_test` not running still stands.

Comment (compnerd) [reply utils/swift_build_support/swift_build_support/products/tensorflow.py]: https://github.com/tensorflow/swift-apis/pull/781 is the associated change for the swift-apis repository.

Comment (asuhan) [reply utils/swift_build_support/swift_build_support/products/tensorflow.py]: With that change, now it doesn't even build for me, it's trying to find libx10.so under bazel-bin/tensorflow.

**Comment by asuhan** [comment]
Here's the errors I get when running ops_test: https://pastebin.com/KXfNRrcE. Having libtensorflow.so and libx10.so separate is a problem because of operator registration and ops_test (and other use cases) needs to import both.

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Comment by compnerd** [comment]
@swift-ci please test tensorflow

**Review by asuhan — APPROVED** [review]

**Review by asuhan — APPROVED** [review]
Please clean up the description before merging.

**Pull request closed without merging**